client/state/selectors/get-notification-unseen-count.js

```diff
@@ -0,0 +1 @@
+export default state => state.notifications;
```

Comment:
supposing we wanted an indication of the boolean value of this count then here is a way I might approach it…

```js
// state/selectors/has-unseen-notifications.js
import getUnseenCount from 'state/selectors/get-notifications-unseen-count';
import { getCurrentUser as getUser } from 'state/current-user/selectors';

const hasUnseenNotifications = ( state ) => {
	const unseenCount = getUnseenCount( state );
	if ( null !== unseenCount ) {
		return unseenCount > 0;
	}

	const user = getUser( state );
	if ( null !== user ) {
		return !! user.has_unseen_notes;
	}

	return false;
};

export default hasUnseenNotifications;
```

Reply:
```js
// state/selectors/test/has-unseen-notifications.js
import has from 'state/selectors/has-unseen-notifications';

const user = hasUnseen => ( {
	currentUser: { id: 1234 },
	users: {
		items: {
			1234: {
				has_unseen_notes: hasUnseen,
			}
		}
	}
} );

test( 'assumes nothing on empty state', () => {
	expect( has( {} ) ).toBeNull();
} );

test( 'follows unseen count', () => {
	expect( has( { notifications: 4 } ) ).toBeTruthy();
	expect( has( { notifications: 0 } ) ).toBeFalsey();
} );

test( 'prefers unseen count over user info', () => {
	expect( has( { notifications: 5, ...user( true ) ) ).toBeTruthy();
} );

test( 'falls back to user info', () => {
	expect( has( { notifications: null, ...user( false ) } ) ).toBeFalsey();
	expect( has( { notifications: null, ...user( true ) } ) ).toBeTruthy();
} );
```

so this obviously demonstrates weakness with how much we have to mock - what could we do? 🤔 

let's inject!

```js
// state/selectors/has-unseen-notifications.js
import getUnseenCount from 'state/selectors/get-notifications-unseen-count';
import { getCurrentUser } from 'state/current-user/selectors';

export const hasUnseenNotifications = getUser => state => {
	const unseenCount = getUnseenCount( state );
	if ( null !== unseenCount ) {
		return unseenCount > 0;
	}

	const user = getUser( state );
	if ( null !== user ) {
		return !! user.has_unseen_notes;
	}

	return null;
};

// export default with getCurrentUser from state
export default hasUnseenNotifications( getCurrentUser );
```

now rewrite tests with dependency injection

```js
// state/selectors/test/has-unseen-notifications.js
import { hasUnseenNotifications as has } from 'state/selectors/has-unseen-notifications';

test( 'assumes nothing on empty state', () => {
	expect( has( noop )( {} ) ).toBeNull();
} );

test( 'follows unseen count', () => {
	expect( has( noop )( { notifications: 4 } ) ).toBeTruthy();
	expect( has( noop )( { notifications: 0 } ) ).toBeFalsey();
} );

test( 'prefers unseen count over user info', () => {
	expect( has( () => false )( { notifications: 5, ...user( true ) ) ).toBeTruthy();
} );

test( 'falls back to user info', () => {
	expect( has( () => false )( { notifications: null } ) ).toBeFalsey();
	expect( has( () => true )( { notifications: null } ) ).toBeTruthy();
} );
```